packages/sak-behandling-velger/src/components/BehandlingPicker.tsx

```diff
@@ -65,22 +103,129 @@ interface OwnProps {
+  const [valgtBehandlingId, setValgtBehandlingId] = useState(behandlingId);
+  const previousBehandlingId = usePrevious(behandlingId);
+  const [søknadsperioder, setSøknadsperioder] = useState<Array<PerioderMedBehandlingsId>>([]);
+  const [activeFilters, setActiveFilters] = useState([]);
+
+  useEffect(() => {
+    if (!previousBehandlingId && behandlingId) {
+      setValgtBehandlingId(behandlingId);
+    }
+  }, [behandlingId]);
+
+  useEffect(() => {
+    const perioder = [];
+    Promise.all(
+      behandlinger.map(behandling =>
+        axios
+          .get(behandling.links.find(link => link.rel === 'behandling-perioder-årsak').href)
+          .then(response => ({ data: response.data, id: behandling.id })),
+      ),
+    ).then((responses: { data: BehandlingPerioder; id: number }[]) => {
+      responses.forEach(({ data, id }) => {
+        perioder.push({ id, perioder: data.perioderTilVurdering });
+      });
+      setSøknadsperioder(perioder);
+    });
+  }, []);
+
+  const valgtBehandling = valgtBehandlingId
+    ? behandlinger.find(behandling => behandling.id === valgtBehandlingId)
+    : null;
+
+  const updateFilter = valgtFilter => {
+    if (activeFilters.includes(valgtFilter)) {
+      setActiveFilters(activeFilters.filter(v => v !== valgtFilter));
+    } else {
+      setActiveFilters(activeFilters.concat([valgtFilter]));
+    }
+  };
+
+  const getFilterListe = () => {
+    const filterListe = [];
+    behandlinger.forEach(behandling => {
+      if (!filterListe.some(filter => filter.value === behandling.type.kode)) {
+        filterListe.push({
+          value: behandling.type.kode,
+          label: getBehandlingNavn(behandling, getKodeverkFn, intl),
+        });
+      }
+    });
+    return filterListe;
+  };
+
+  return (
+    <div className={styles.behandlingPicker}>
+      {valgtBehandlingId && (
+        <Tilbakeknapp className={styles.backButton} onClick={() => setValgtBehandlingId(undefined)}>
+          <FormattedMessage id="Behandlingspunkt.Behandling.SeAlle" />
+        </Tilbakeknapp>
+      )}
 
+      {!valgtBehandlingId && (
+        <>
+          <div className={styles.headerContainer}>
+            <Undertittel>
+              <FormattedMessage
+                id="Behandlingspunkt.VelgBehandling"
+                values={{ antallBehandlinger: behandlinger.length }}
+              />
+            </Undertittel>
+            <BehandlingFilter
+              filters={getFilterListe()}
+              activeFilters={activeFilters}
+              onFilterChange={updateFilter}
+              text={intl.formatMessage({ id: 'Behandlingspunkt.Filtrer' })}
+            />
+          </div>
+          <ul className={styles.behandlingList}>
+            {noExistingBehandlinger && (
+              <Normaltekst>
+                <FormattedMessage id="BehandlingList.ZeroBehandlinger" />
+              </Normaltekst>
+            )}
+            {!noExistingBehandlinger &&
+              renderListItems(
```

Comment:
blir det mer clean med destructuring her?

Reply:
hmm, hva mener du?

Reply:
sende inn ett objekt som destruktureres i stedet for at man må ha parameterne i riktig rekkefølge

Reply:
Sånn ja. Kan teste det